src/commands/scanner/run.ts

```diff
@@ -119,10 +124,22 @@ export default class Run extends ScannerCommand {
 			target.push(this.args.file);
 		}
 		const targetPaths = target.map(path => normalize(untildify(path)).replace(/['"]/g, ''));
-		const output = await ruleManager.runRulesMatchingCriteria(filters, targetPaths, format);
+		const engineOptions = this.gatherEngineOptions();
+		const output = await ruleManager.runRulesMatchingCriteria(filters, targetPaths, format, engineOptions);
 		return this.processOutput(output);
 	}
 
+	/**
+	 * Gather a map of options that will be passed to the RuleManager without validation.
+	 */
+	private gatherEngineOptions(): Map<string, string> {
+		const options = new Map();
+		if (this.flags.tsconfig) {
+			const tsconfig = normalize(untildify(this.flags.tsconfig));
+			options.set(TYPESCRIPT_ENGINE_OPTIONS.TSCONFIG, tsconfig);
```

Comment:
Instead of gathering these options and overrides at the top-level and passing them all the way through, could we possibly use them here to set something in the `Config.ts` file, and then the typescript strategy can look at the config for its engine?
If that doesn't work, what other options  do we have for making this work in a way that preserves the Engine abstraction we've been using?

Reply:
I totally agree, this starts to leak. The reason I think it's passable for 2.0, is that fixing it in 3.0 would not require a user visible change. It's an implementation detail.

Using Config.ts would still require the run command to know which engine to configure. Using the file as an intermediary breaks down if a single machine would run multiple instances of the scanner simultaneously.

3.0 is a good time to revisit the abstraction. the RuleEngine#run command currently takes parameters that are exclusive to either the ESLint or PmdEngines (RuleGroup vs. Rule). It would be good to come up with a pattern that doesn't have unused parameters to the method signature as well.

Reply:
This might be an interesting idea for avoiding passing stuff through all of the layers https://medium.com/@tabu_craig/nodejs-and-thread-local-storage-eb2c1a24881 another option would be to pass an object to the methods instead of individual parameters, adding a new property to the object insulates the layers in between from needing to change each time.